src/components/modules/api/toolbar.ts

```diff
@@ -31,4 +32,36 @@ export default class ToolbarAPI extends Module {
   public close(): void {
     this.Editor.Toolbar.close();
   }
+
+  /**
+   * Toggles Block Setting of the current block
+   *
+   * @param {boolean} openingState —  opening state of Block Setting
+   */
+  public toggleBlockSettings(openingState?: boolean): void {
+    if (!this.Editor.BlockManager.currentBlockIndex) {
+      _.logLabeled('Could\'t toggle the Toolbar because there is no block selected ', 'warn');
+
+      return;
+    }
+
+    if (!this.Editor.Toolbar.opened) {
```

Comment:
I think this if statement should be moved into: 

```
if (canOpenBlockSettings) {
  this.Editor.BlockSettings.open();
}
```

Reply:
Done